ide/coqide/protocol/xmlprotocol.ml

```diff
@@ -1021,9 +1013,12 @@ let document to_string_fmt =
     (to_string_fmt (constructor "call" "C" [PCData "a"]));
   Printf.printf "A response carrying output b can either be:\n\n%s\n\n"
     (to_string_fmt (of_value (fun _ -> PCData "b") (Good ())));
-  Printf.printf "or:\n\n%s\n\nwhere the attributes loc_s and loc_c are optional.\n"
+  Printf.printf "or:\n\n%s\n\n"
     (to_string_fmt (of_value (fun _ -> PCData "b")
-      (Fail (Stateid.initial,Some (15,34), Pp.str "error message"))));
+      (Fail
+         (Stateid.initial,
+          Some (Loc.{fname=ToplevelInput; line_nb=4; bol_pos=0; line_nb_last=6; bol_pos_last=10; bp=15; ep=34}),
```

Comment:
Better, but still has various magic numbers and not obvious how they were determined.  But not essential to clean up this bit in this PR.

Reply:
That's not magic numbers, it is an arbitrary example for documenting the protocol (it is in function `document`).